oaknational/Oak-Web-Application
# LESQ-1005: Can't find your school form

## Description

Music year: 1973

- Adds `ManualEntryComponent`
- Adds `SchoolDetailsInputComponent`
- Amends schema definition for school selection form
- Adds buttons to switch between forms
- Adds error states / components for forms
- onSubmit registers data to be connected to hubspot form

## Issue(s)

Do we want the values in the form to persist or refresh when changing page?

Acceptance criteria
--
- [ ] Add button to the school picker “Enter manually” that changes input options to “School name (required)” and “School address (required)” free text input
- [ ] Button “Select from dropdown” to take you back to school picker
- [ ] Errors if text inputs are empty


Fixes #LESQ-1005

## How to test

1. Go to https://deploy-preview-2709--oak-web-application.netlify.thenational.academy/onboarding
2. Click on yes, when posed with the do you work in a school question
3. You should see the following page
![Screenshot 2024-08-16 at 13 49 51](https://github.com/user-attachments/assets/4418c0f8-494c-4ea9-8dbc-f15daaf5d788)
4. Enter a school to use the school picker, the button should become enabled when school picker has a value and continue
4b. Select enter manually button
5. School name and school address input boxes should render, fill both of them for continue button to be available
![Screenshot 2024-08-16 at 13 51 13](https://github.com/user-attachments/assets/7796c523-1b05-488f-a5cf-187414221de4)
6. If you select from dropdown, returns to SchoolPicker component


## Screenshots



https://github.com/user-attachments/assets/7d8c9fda-5eeb-4c6c-8292-4562b4073060




## Checklist
